Description: read policies from a snap-aware path (typically /var/snap/chromium/current/)
Author: Olivier Tilloy <olivier.tilloy@canonical.com>
Bug: https://bugs.launchpad.net/ubuntu/+source/chromium-browser/+bug/1714244

--- a/chrome/common/chrome_paths.cc
+++ b/chrome/common/chrome_paths.cc
@@ -493,6 +493,13 @@ bool PathProvider(int key, base::FilePat
       break;
 #if defined(OS_POSIX) && !defined(OS_MACOSX) && !defined(OS_OPENBSD)
     case chrome::DIR_POLICY_FILES: {
+      std::unique_ptr<base::Environment> env(base::Environment::Create());
+      std::string snap_data;
+      if (env->GetVar("SNAP_DATA", &snap_data)) {
+        cur = base::FilePath(snap_data);
+        cur.Append(FILE_PATH_LITERAL("policies"));
+        break;
+      }
 #if BUILDFLAG(GOOGLE_CHROME_BRANDING)
       cur = base::FilePath(FILE_PATH_LITERAL("/etc/opt/chrome/policies"));
 #else
